src/neuroconv/utils/json_schema.py

```diff
@@ -48,6 +53,68 @@ def get_base_schema(tag=None, root=False, id_=None, **kwargs) -> dict:
     return base_schema
 
 
+def get_pydantic_model_from_method_signature(function: callable, exclude: Optional[List[str]] = None) -> BaseModel:
+    """
+    Generate a Pydantic model from the signature of an arbitrary function which uses keywords with annotation typing.
+
+    Parameters
+    ----------
+    function : callable
+        Any function that uses keywords with annotation types.
+    exclude : list of strings, default: None
+        A list of keys to exclude from the model but are present in the function signature.
+        Can be useful when skipping reserved names such as 'self' in methods of a class.
+
+    Returns
+    -------
+    model : instance of a pydantic model
+    """
+    # The default model of the parent 'ValidatedFunction' includes many fields we do not need
+    # but typical model-copying methods only work on fully instantiated models (which also triggers validation),
+    # whereas the model of this decorator class works more as a class method
+    class CustomValidatedFunction(ValidatedFunction):
+        def __init__(self, function: callable, exclude: Optional[List[str]] = None):
+            config = dict(
+                arbitrary_types_allowed=True,  # Needed for NWBFile annotations and anything from Numpy
+            )
+
+            super().__init__(function=function, config=config)  # Sets various attributes other methods assume exist
+
+            parameters: Dict[str, Parameter] = signature(function).parameters
+            fields: Dict[str, Tuple[Any, Any]] = dict()
+
+            if exclude is not None:
+                parameters = {k: v for k, v in parameters.items() if k not in exclude}
+
+            for name, parameter in parameters.items():
+                default = ... if parameter.default is Parameter.empty else parameter.default
+                fields[name] = parameter.annotation, default
+
+            self.create_model(
+                fields=fields,
+                takes_args=False,
+                takes_kwargs=True,
+                config=config,
+            )
+
+    return CustomValidatedFunction(function=function, exclude=exclude).model
+
+
+def get_schema_from_pydantic_model(model: BaseModel) -> Dict[str, Any]:
+    """
+    Return the JSON schema of a Pydantic model.
+
+    Parameters
+    ----------
+    model : any pydantic model
+
+    Returns
+    -------
+    json_schema : dict
+    """
+    return model_schema(model)
```

Comment:
I don't think this function is necessary. We can just call `model_schema`, right?